Added icons for 6 new resource types in JsonOutlineProvider

nilshedstrom opened the pull request:
Added icons for the following resource types in the JsonOutlineProvider

- Microsoft.Web/sites
- config
- Microsoft.Insights/components
- Microsoft.KeyVault/vaults
- Microsoft.KeyVault/vaults/secrets
- Microsoft.Cdn/profiles

The new icons were downloaded from https://www.microsoft.com/en-us/download/details.aspx?id=41937

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/microsoft/vscode-azurearmtools?pullRequest=417) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/vscode-azurearmtools?pullRequest=417) before we can accept your contribution.<br/><br/>:x: nilshedstrom [sign now](https://cla.opensource.microsoft.com/microsoft/vscode-azurearmtools?pullRequest=417)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/microsoft/vscode-azurearmtools?pullRequest=417) it.</sub>

StephenWeatherford commented:
@nilshedstrom Thanks!  They all look great, except for "config", which is difficult to see on default dark theme:
![image](https://user-images.githubusercontent.com/6913354/70840357-8fb5ae00-1dc6-11ea-9d38-3bb9581bb67c.png)
and is a different size than the others:
![image](https://user-images.githubusercontent.com/6913354/70840368-9d6b3380-1dc6-11ea-9823-376c634c624d.png)

Did this icon come from a different place?  Thx.

nilshedstrom commented:
> @nilshedstrom Thanks! They all look great, except for "config", which is difficult to see on default dark theme:
> ![image](https://user-images.githubusercontent.com/6913354/70840357-8fb5ae00-1dc6-11ea-9d38-3bb9581bb67c.png)
> and is a different size than the others:
> ![image](https://user-images.githubusercontent.com/6913354/70840368-9d6b3380-1dc6-11ea-9823-376c634c624d.png)
> 
> Did this icon come from a different place? Thx.

There is another settings.svg in Azure Icons which is gray but that one did not work with vscode for some reason (displayed as a single pixel).
I changed the icon to the one for Application Configuration which has mor colour and does not look like the parameters icon.

nilshedstrom commented:
I do not see how my change caused the test TLE colorization array-access.INVALID.json to fail on the Windows build (but passed on Linux and Mac). It seems like it failed on the 2019-12-03 too.

nilshedstrom commented:
I added an icon for functions (which should solve issue #421).
Refactored the existing icon tests to one (should be a bit faster).
Added tests for function icons.

StephenWeatherford commented:
We'll be releasing very soon, hopefully Monday.
